### PR TITLE
Refactor DodgemsCarWouldCollideAt to return optional

### LINKED ISSUE #16683: Refactor Vehicle::DodgemsCarWouldCollideAt to return optional
Use std::optional to return the EntityId

_Originally posted by @Gymnasiast in https://github.com/OpenRCT2/OpenRCT2/pull/16640#discussion_r810058097_

### PR DESCRIPTION
**Pull request by GalBr**
Closes  #16683

### CODE REVIEW

File: src/openrct2/ride/Vehicle.cpp [hunk 1]
```diff
@@ -6275,7 +6275,8 @@ int32_t Vehicle::UpdateMotionDodgems()
         location.x += Unk9A36C4[oldCollisionDirection + 1].x;
         location.y += Unk9A36C4[oldCollisionDirection + 1].y;
 
-        if (!DodgemsCarWouldCollideAt(location, &collideSprite))
+        collideSprite = DodgemsCarWouldCollideAt(location);
+        if (collideSprite == std::nullopt)
```
Comment (tupaschoal) [inline src/openrct2/ride/Vehicle.cpp:6279]:
```suggestion
        if (collideSprite = DodgemsCarWouldCollideAt(location); collideSprite == std::nullopt)
```

File: src/openrct2/ride/Vehicle.cpp [hunk 2]
```diff
@@ -6300,8 +6301,11 @@ int32_t Vehicle::UpdateMotionDodgems()
             location.x += Unk9A36C4[direction].x;
             location.y += Unk9A36C4[direction].y;
 
-            if (DodgemsCarWouldCollideAt(location, &collideSprite))
+            collideSprite = DodgemsCarWouldCollideAt(location);
+            if (collideSprite != std::nullopt)
```
Comment (tupaschoal) [inline src/openrct2/ride/Vehicle.cpp:6305]:
```suggestion
            if (collideSprite = DodgemsCarWouldCollideAt(location); collideSprite != std::nullopt)
```

**Review by tupaschoal — COMMENTED** [review]
Seems mostly alright

**Comment by GalBr** [comment]
I updated the branch with the suggested changes and replaced the comparisons against `std::nullopt` to `.has_value()`.

File: src/openrct2/ride/Vehicle.cpp [hunk 3]
```diff
@@ -6262,7 +6262,7 @@ int32_t Vehicle::UpdateMotionDodgems()
         }
     }
 
-    auto collideSprite = EntityId::GetNull();
+    std::optional<EntityId> collideSprite = EntityId::GetNull();
```
Comment (ZehMatt) [inline src/openrct2/ride/Vehicle.cpp:6265]: The value will be overwritten, just leave it valueless.

**Review by ZehMatt — CHANGES_REQUESTED** [review]
Minor thing otherwise fine with me.

File: src/openrct2/ride/Vehicle.cpp (hunk 3 above)
Comment (GalBr) [reply src/openrct2/ride/Vehicle.cpp]: You are correct, done.

**Review by tupaschoal — COMMENTED** [review]
There's a merge conflict, can you please tend to it and let us know when you're done, please @GalBr ?

**Comment by GalBr** [comment]
@tupaschoal  I fixed the merge conflict.

**Review by duncanspumpkin — APPROVED** [review]

**Pull request merged**